tracing/tracing.go

```diff
@@ -6,58 +6,26 @@ package tracing
 
 import (
 	"context"
-	"io"
 
 	"github.com/opentracing/opentracing-go"
 	"github.com/pkg/errors"
 	jaeger "github.com/uber/jaeger-client-go"
-	jaegercfg "github.com/uber/jaeger-client-go/config"
-	jaegerprom "github.com/uber/jaeger-lib/metrics/prometheus"
+	"go.opentelemetry.io/otel/trace"
 )
 
-// ErrInvalidConfiguration is an error to notify client to provide valid trace report agent or config server
 var (
+	// ErrBlankTraceConfiguration is an error to notify client to provide valid trace report agent or config server
 	ErrBlankTraceConfiguration = errors.New("no trace report agent, config server, or collector endpoint specified")
 )
 
-// installJaeger registers Jaeger as the OpenTracing implementation.
-func installJaeger(serviceName string, cfg *jaegercfg.Configuration, options ...jaegercfg.Option) (io.Closer, error) {
-	metricsFactory := jaegerprom.New()
-
-	// put the metricsFactory earlier so provided options can override it
-	opts := append([]jaegercfg.Option{jaegercfg.Metrics(metricsFactory)}, options...)
-
-	closer, err := cfg.InitGlobalTracer(serviceName, opts...)
-	if err != nil {
-		return nil, errors.Wrap(err, "could not initialize jaeger tracer")
-	}
-	return closer, nil
-}
-
-// NewFromEnv is a convenience function to allow tracing configuration
-// via environment variables
-//
-// Tracing will be enabled if one (or more) of the following environment variables is used to configure trace reporting:
-// - JAEGER_AGENT_HOST
-// - JAEGER_SAMPLER_MANAGER_HOST_PORT
-func NewFromEnv(serviceName string, options ...jaegercfg.Option) (io.Closer, error) {
-	cfg, err := jaegercfg.FromEnv()
-	if err != nil {
-		return nil, errors.Wrap(err, "could not load jaeger tracer configuration")
-	}
-
-	if cfg.Sampler.SamplingServerURL == "" && cfg.Reporter.LocalAgentHostPort == "" && cfg.Reporter.CollectorEndpoint == "" {
-		return nil, ErrBlankTraceConfiguration
-	}
-
-	return installJaeger(serviceName, cfg, options...)
-}
-
 // ExtractTraceID extracts the trace id, if any from the context.
 func ExtractTraceID(ctx context.Context) (string, bool) {
 	if tid, _, ok := extractJaegerContext(ctx); ok {
```

Comment:
nit: maybe we could reuse `SpanFromContext` in the this and following function? Isn't extraction from context now implemented separately for otrace and otel and then in `SpanFromContext` for both at the same time?

Reply:
It would actually look longer:
```go
// ExtractTraceID extracts the trace id, if any from the context.
func ExtractTraceID(ctx context.Context) (string, bool) {
	otelSpan, opentracingSpan, _ := SpanFromContext(ctx)
	if otelSpan != nil {
		if sc := otelSpan.SpanContext(); sc.IsValid() {
			return sc.TraceID().String(), true
		}
	}
	if opentracingSpan != nil {
		if sctx, ok := opentracingSpan.Context().(jaeger.SpanContext); ok {
			return sctx.TraceID().String(), true
		}
	}
	return "", false
}
```
I'd rather leave it as is, and make everything cleaner once we get rid of opentracing.